docs-src/guides/add-new-blockchain.md

```diff
@@ -0,0 +1,67 @@
+This has proven to be useful fir dApp personalization: one sees familiar names instead of `0xgibberish`.
+
+Below one additional process is mentioned: validation. It checks if _proof-of-ownership_ in the link is formally correct,
+i.e., a well-known payload is really signed by the account that is declared in the link.
+
+## Adding a new blockchain
+
+To add a new blockchain one have to do both linking and validation sections. 
+
+### Linking
+
+We use [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) to represent accounts in a blockchain agnostic way.
+If the blockchain you want to add isn't already part of the CAIP standards you should make sure to add it there.
+
+To add a new blockchain, one has to implement a new class implementing [AuthProvider](https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/blockchain-utils-linking/src/auth-provider.ts), put it
+into [`@ceramicnetwork/blockchain-utils-linking`](https://github.com/ceramicnetwork/js-ceramic/tree/develop/packages/blockchain-utils-linking) package and export it.
+See existing auth providers for inspiration.
+
+Auth provider is expected to be mainly called in a web browser as part of [3id-connect](https://github.com/3box/3id-connect) flow.
+Auth provider sits between 3id-connect (or 3ID DID Provider) and your blockchain account provider. In case of Ethereum,
+it might be MetaMask. Auth provider is responsible mainly for:
+- authentication (`#authenticate`): provide entropy,
+- creating link (`#createLink`): create a LinkProof object which associates the specified AccountID with the DID.
+
+Auth provider is expected to know which blockchain account it currently serves. It reports it via `#accountId`.
+To reuse same internal settings, e.g. connection to blockchain provider, but with a different account,
+auth provider should have `#withAddress` method.
+
+To sum it all up, to add a new blockchain for linking:
+- add new class implementing `AuthProvider` interface to `@ceramicnetwork/blockchain-utils-linking` package,
+- its `authenticate` method should ask blockchain provider for an authentication signature and return 32 bytes
+  in form of `0x`-prefixed hex string,
+- its `createLink` method should ask blockchain provider for linking signature and return `LinkProof` data structure,
+- its `accountId` method should return currently used account in CAIP-10 format,
+- its `withAccount` method should return a new instance of auth provider that serves a new account.
+
+### Validation
+
+It is a counterpart of [#linking] that checks if signature contained in `LinkProof` corresponds to declared account.
+
+To add a new blockchain:
+- add a new file named after your blockchain to `@ceramicnetwork/blockchain-utils-validation` package,
+- this file should expose `BlockchainHandler` structure, having:
+    - CAIP-10 namespace for your blockchain,
```

Comment:
```suggestion
    - the CAIP-10 name of your blockchain
```

Reply:
It should be "`CAIP-2` namespece" actually!

Reply:
CAIP-10 descripes a blockchain accountId based on caip-2

Reply:
Is a `CAIP-2` actually a "namespace"?  Should this just be "the CAIP-2 name of your blockchain"?

Reply:
it is CAIP-2 _namespace_ indeed, as per spec https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md

Reply:
Are we referring to *namespace* (without *reference*) or *chain_id* here?

Reply:
> it is CAIP-2 _namespace_ indeed, as per spec https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md

Ah cool, good to know!

Reply:
Namespace. Definitely.

Reply:
> Are we referring to _namespace_ (without _reference_) or _chain_id_ here?

Since CAIP-10 links are to a specific address, I imagine that must be on a specific chain, so this should probably actually be `chain_id` no?

Reply:
chain id is `namespace` + `reference`.

Assumption here is blockchains under the same namespace share the same signature mechanism, so linking/validation are scoped per namespace.

Reply:
Makes sense, thanks!